src/util.cpp

```diff
@@ -108,6 +108,33 @@ std::string get_process_comm(pid_t pid)
     }
 }
 
+std::vector<std::string> get_process_cmdline(pid_t pid)
+{
+    try
+    {
+        std::vector<std::string> cmdline;
+        boost::filesystem::ifstream s{ boost::filesystem::path{ "/proc" } / std::to_string(pid) /
+                                       "cmdline" };
+
+        // /proc/<pid>/cmdline is a NUL-delimited list of arguments
+        for (std::string arg; std::getline(s, arg, '\0');)
```

Comment:
in the second iteration, you sort of dereferencing a move-from `arg` here. I'm scared